isis/src/clipper/objs/ClipperPushBroomCamera/ClipperPushBroomCamera.cpp

```diff
@@ -0,0 +1,129 @@
+   *
+   * @param Cube &cube Clipper EIS image.
+   */
+   ClipperPushBroomCamera::ClipperPushBroomCamera(Cube &cube) : LineScanCamera(cube) {
+
+     m_spacecraftNameLong = "Europa Clipper";
+     m_spacecraftNameShort = "Clipper";
+
+     int frameCode = naifIkCode();
+
+     // ik values for NAC: -159101(general) and filters between -159121 and -159131
+     if (frameCode == -159101 || (frameCode > -159132 && frameCode < -159120)) {
+       m_instrumentNameLong  = "Europa Imaging System Push Broom Narrow Angle Camera";
+       m_instrumentNameShort = "EIS-PBNAC";
+     }
+     // ik values for WAC: -159102(general) and filters between -159141 and -159151
+     else if (frameCode == -159102 || (frameCode > -159152 && frameCode < -159140)) {
+       m_instrumentNameLong  = "Europa Imaging System Push Broom Wide Angle Camera";
+       m_instrumentNameShort = "EIS-PBWAC";
+     }
+     else {
+       QString msg = "Unable to construct Clipper Push Broom camera model. "
+                     "Unrecognized NaifFrameCode [" + toString(frameCode) + "].";
+       throw IException(IException::User, msg, _FILEINFO_);
+     }
+
+     NaifStatus::CheckErrors();
+
+     SetFocalLength();
+     SetPixelPitch();
+
+     Pvl &lab = *cube.label();
+     PvlGroup &inst = lab.findGroup("Instrument", Pvl::Traverse);
+     double lineRate = ((double) inst["LineExposureDuration"]) / 1000;
+     QString startTime = inst["StartTime"];
+     iTime etStart(startTime);
+
+     // set up detector map
+     new LineScanCameraDetectorMap(this, etStart.Et(), lineRate);
+
+     // Set up focal plane map
+     CameraFocalPlaneMap *focalMap = new CameraFocalPlaneMap(this, naifIkCode());
+     focalMap->SetDetectorOrigin(2048.5, 1024.5);
+
+     // Set up distortion map
+     new CameraDistortionMap(this);
+     // TODO: set distortion
+
+     // Set up the ground and sky map
+     new LineScanCameraGroundMap(this);
+     new LineScanCameraSkyMap(this);
+
+     LoadCache();
+     NaifStatus::CheckErrors();
+   }
+
+
+   /**
+    * Destructor for a ClipperPushBroomCamera object.
+    */
+   ClipperPushBroomCamera::~ClipperPushBroomCamera() {
+   }
+
+
+   /**
+    * CK frame ID - Instrument Code from spacit run on CK
+    *
+    * @return @b int The appropriate instrument code for the "Camera-matrix"
+    *         Kernel Frame ID
+    */
+   int ClipperPushBroomCamera::CkFrameId() const {
+     return (-159000);
+   }
+
+
+   /**
+    * CK Reference ID - J2000
+    *
+    * @return @b int The appropriate instrument code for the "Camera-matrix"
+    *         Kernel Reference ID
+    */
+   int ClipperPushBroomCamera::CkReferenceId() const {
+     return (-159010);
+   }
+
+
+   /**
+    * SPK Reference ID - J2000
+    *
+    * @return @b int The appropriate instrument code for the Spacecraft
+    *         Kernel Reference ID
+    */
+   int ClipperPushBroomCamera::SpkReferenceId() const {
+     return (1);
+   }
```

Comment:
Do we still need to update this based on the spacit output?

Reply:
This are now accurate to spacit output.